exposestrategy/ingress.go

```diff
@@ -278,18 +246,69 @@ func (s *IngressStrategy) Add(svc *api.Service) error {
 		}
 	}
 
-	if createIngress {
-		_, err := s.client.Ingress(ingress.Namespace).Create(ingress)
+	existing, err := cli.Ingress(svc.Namespace).Get(appName)
+	if err != nil {
+		if !apierrors.IsNotFound(err) {
+			return "", errors.Wrapf(err, "could not check for existing ingress %s/%s", svc.Namespace, appName)
+		}
+		_, err := cli.Ingress(ingress.Namespace).Create(ingress)
 		if err != nil {
-			return errors.Wrapf(err, "failed to create ingress %s/%s", ingress.Namespace, ingress.Name)
+			return "", errors.Wrapf(err, "failed to create ingress %s/%s", ingress.Namespace, ingress.Name)
```

Comment:
Consider: This is confusing, my concern is you'll lose an error here.  I can see alot of the code is existing.